### PR TITLE
add docs for enforce mpp

### PR DESCRIPTION
**Pull request by LittleFall**
### What is changed, added or deleted? (Required)

add docs for enforce mpp

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)

don't need cherry-pick now

### What is the related PR or file link(s)?

- This PR is translated from:  need translate
- Other reference link(s):https://github.com/pingcap/tidb/pull/24516

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -181,7 +181,12 @@ mysql> SELECT * FROM t1;
 
 - 作用域：SESSION | GLOBAL
 - 默认值：ON（表示开启）
-- 这个变量用于控制是否使用 TiFlash 的 MPP 模式执行查询。开启后 TiDB 会通过优化器自动判断是否选择 MPP 执行。MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。
+- 这个变量用于控制是否使用 TiFlash 的 MPP 模式执行查询，可以设置的值包括：
+    - 0 或 OFF，代表从不使用 mpp 模式；
+    - 1 或 ON，代表由优化器根据代价估算选择是否使用 mpp 模式；（默认）
+    - 2 或 ENFORCE，代表无视代价估算，尽可能使用 mpp 模式；
+
+MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。
```
Comment (TomShawn) [inline system-variables.md:189]:
```suggestion
- 这个变量用于控制是否使用 TiFlash 的 MPP 模式执行查询，可以设置的值包括：
    - 0 或 OFF，代表从不使用 MPP 模式
    - 1 或 ON，代表由优化器根据代价估算选择是否使用 MPP 模式（默认）
    - 2 或 ENFORCE，代表无视代价估算，尽可能使用 MPP 模式

MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。
```

File: tiflash/use-tiflash.md [hunk 2]
```diff
@@ -246,7 +246,13 @@ round without fraction, cast(int as decimal), date_add(datetime, int), date_add(
 
 ## 使用 MPP 模式
 
-TiFlash 支持 MPP 模式的查询执行，即在计算中引入跨节点的数据交换（data shuffle 过程）。MPP 模式默认开启，如需关闭可将全局/会话变量 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 的值设为 `0` 或 `OFF`：
+TiFlash 支持 MPP 模式的查询执行，即在计算中引入跨节点的数据交换（data shuffle 过程）。默认由优化器自动选择是否使用 MPP 模式，可以通过修改变量 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 的值来更改选择策略，取值包括：
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:249]:
```suggestion
TiFlash 支持 MPP 模式的查询执行，即在计算中引入跨节点的数据交换（data shuffle 过程）。默认由优化器自动选择是否使用 MPP 模式，可以通过修改变量 [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 的值来更改选择策略，变量取值包括：
```

File: tiflash/use-tiflash.md [hunk 3]
```diff
@@ -246,7 +246,13 @@ round without fraction, cast(int as decimal), date_add(datetime, int), date_add(
...
+
+- 0 或 OFF，代表从不使用 MPP 模式；
+- 1 或 ON，代表由优化器根据代价估算选择是否使用 MPP 模式；（默认）
+- 2 或 ENFORCE，代表无视代价估算，尽可能使用 MPP 模式；
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:253]:
```suggestion
- 0 或 OFF，代表从不使用 MPP 模式
- 1 或 ON，代表由优化器根据代价估算选择是否使用 MPP 模式（默认）
- 2 或 ENFORCE，代表无视代价估算，尽可能使用 MPP 模式
```

File: tiflash/use-tiflash.md [hunk 4]
```diff
@@ -246,7 +246,13 @@ round without fraction, cast(int as decimal), date_add(datetime, int), date_add(
...
+
+举例来说，如果不想使用 MPP，可以通过如下语句来设置：
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:255]:
```suggestion
例如，如果你不想使用 MPP 模式，可以通过以下语句来设置：
```

**Review by TomShawn — COMMENTED** [review]
some minor comments

**Comment by qiancai** [comment]
According to @LittleFall, the changes of this PR are for v5.1 and the changes are still draft so not ready for review.

**Comment by TomShawn** [comment]
@LittleFall Any update?

**Review by hanfei1991 — APPROVED** [review]

File: system-variables.md [hunk 5]
```diff
@@ -194,7 +194,21 @@ mysql> SELECT * FROM t1;
...
+    - 1 或 ON，代表由优化器根据代价估算选择是否使用 MPP 模式（默认）
+
+MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数据交换并提供高性能、高吞吐的 SQL 算法。
+
+### `tidb_enforce_mpp` <span class="version-mark">从 v5.1 版本开始引入</span>
+
+- 作用域：SESSION
+- 默认值：OFF（表示关闭）
+- 这个变量用于控制是否忽略优化器代价估算，强制使用 TiFlash 的 MPP 模式执行查询，可以设置的值包括：
+    - 0 或 OFF，代表将不会强制使用 mpp 模式（默认）
```
Comment (TomShawn) [inline system-variables.md:208]:
```suggestion
    - 0 或 OFF，代表不强制使用 MPP 模式（默认）
```

File: tiflash/use-tiflash.md [hunk 6]
```diff
@@ -254,12 +254,72 @@ TiFlash 支持部分算子的下推，支持的算子如下：
...
+set @@session.tidb_allow_mpp=0;
+```
+
+如果想要通过优化器代价估算来智能选择是否使用 MPP（是默认情况），可以通过如下语句来设置：
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=0;
```
[the remark this answers is not in the document]
Comment (LittleFall) [reply tiflash/use-tiflash.md:284]: 它们都可以生效；一般on/off表示是否打开，但是实际修改变量时一般取值 1/0.

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**